src/blob_file_builder.cc

```diff
@@ -6,18 +6,38 @@ namespace titandb {
 BlobFileBuilder::BlobFileBuilder(const TitanDBOptions& db_options,
                                  const TitanCFOptions& cf_options,
                                  WritableFileWriter* file)
-    : cf_options_(cf_options),
+    : builder_state_(cf_options.blob_file_compression_options.max_dict_bytes > 0
+                         ? BuilderState::kBuffered
+                         : BuilderState::kUnbuffered),
+      cf_options_(cf_options),
       file_(file),
-      encoder_(cf_options_.blob_file_compression) {
+      encoder_(cf_options_.blob_file_compression,
+               cf_options.blob_file_compression_options) {
   BlobFileHeader header;
+  if (cf_options.blob_file_compression_options.max_dict_bytes > 0)
+    header.flags |= BlobFileHeader::kHasUncompressionDictionary;
   std::string buffer;
   header.EncodeTo(&buffer);
   status_ = file_->Append(buffer);
 }
 
 void BlobFileBuilder::Add(const BlobRecord& record, BlobHandle* handle) {
   if (!ok()) return;
+  if (builder_state_ == BuilderState::kBuffered) {
+    std::string rec_str;
```

Comment:
nit: s/rec_str/record_str/

let's use abbreviation less for variables. same below.